### DISCUSSION TITLE
Feature request : override keybindings [shipped]

### DISCUSSION THREAD

**Discussion by CaptainQuirk** [tconbeer/harlequin#422] [Ideas (Feature Requests)]
Hi,

First, I've been waiting for such a tool for quite a time, thanks for the work. It looks really promissing !

It would be really appreciable to be able to override keybindings completely ! Is it on your roadmap ?

Thanks in advance !

**Comment by tconbeer** [comment]
I've had a few requests for vim bindings specifically, which I think is harder (supporting multiple modes); this might be doable. What would your preferred config format look like?

**Comment by CaptainQuirk** [comment]
There may be a need for modes outside the query input area, yes, but I think that first allowing to override basic keybindings would be a good start.

For the format, what about create `[bindings.*]` sections in the toml file where `*` can be _catalog_, _query_ or _general_ ?
For specifying the keys, I dont have a préférence !

**Comment by tconbeer** [comment]
Right now setting key bindings at runtime isn't supported by Textual; I'm going to convert this to a discussion until we get an API for that.

**Comment by CaptainQuirk** [comment]
Well, that's too bad !
I think harlequin is very promising
Unfortunately, without the ability to override keybindings, I'm afraid I won't be able to use it properly !
I'll swing by regularly to see where it's at !
Thanks !

**Comment by kwando** [comment]
vim bindings would be awesome :)

**Comment by tconbeer** [comment]
This will be the next major feature. Please see #541 and provide any feedback on the proposed design.

**Comment by tconbeer** [comment]
This has been shipped in v1.22.0

Vim motions present a suite of challenges that will have to be tackled later; for now, though, you can customize bindings to any single key press. See the new docs: https://harlequin.sh/docs/keymaps/index

**Reply by tconbeer** [reply]
I'm going to lock this discussion; let's continue the Vim conversation in this related discussion: https://github.com/tconbeer/harlequin/discussions/371
